### ISSUE TITLE
Problem with 0.12.0 not creating dynatrace-dynakube-oneagent-unprivileged

### ISSUE THREAD

**Issue by Jaykul** [Dynatrace/dynatrace-operator#1940]
On a clean cluster, the dynatrace-operator is not creating the oneagent-unprivileged service account, but is still using it for the OneAgent DaemonSet, resulting in all OneAgent pods failing to start.

**To Reproduce**

1. On a new cluster, without Dynatrace or the operator (e.g. delete the dynatrace namespace)
2. Install the Dynatrace Operator v0.12.0
3. Deploy Dynakube resource
4. Check for pods in the dynatrace namespace (only the operator, activeGate, and webhooks show up).
6. Check for DaemonSet (see 0 of 3 pods)
7. Look for why. Ran `kubectl get events -n dynatrace`

Found that the state is "FailedCreate" and the following message:

> Error creating: pods "dynasl1-oneagent-" is forbidden: error looking up service account dynatrace/dynatrace-dynakube-oneagent-unprivileged: serviceaccount "dynatrace-dynakube-oneagent-unprivileged" not found

**Expected behavior**

OneAgent should run!

**Additional context**

- We're on AKS
- I installed the operator with OLM
- With v0.11.2 of the operator, and the same config, it creates _additional service accounts_, and everything starts as expected.

### v0.11.2
```
kubectl get serviceaccount -n dynatrace
NAME                                       SECRETS   AGE
default                                    0         8m
dynatrace-activegate                       0         2m35s
dynatrace-dynakube-oneagent-privileged     0         2m37s
dynatrace-dynakube-oneagent-unprivileged   0         2m35s
dynatrace-kubernetes-monitoring            0         2m34s
dynatrace-operator                         0         2m33s
dynatrace-webhook                          0         2m37s
```

### v0.12.0
```
kubectl get serviceaccount -n dynatrace
NAME                                     SECRETS   AGE
default                                  0         9m25s
dynatrace-dynakube-oneagent-privileged   0         9m19s
dynatrace-kubernetes-monitoring          0         9m18s
dynatrace-operator                       0         9m18s
dynatrace-webhook                        0         9m17s
```

My dynakube.yaml:

```yaml
apiVersion: v1
items:
- apiVersion: dynatrace.com/v1beta1
  kind: DynaKube
  metadata:
    annotations:
      feature.dynatrace.com/automatic-kubernetes-api-monitoring: "true"
      feature.dynatrace.com/automatic-kubernetes-api-monitoring-cluster-name: sl1
    creationTimestamp: "2023-07-11T22:07:17Z"
    generation: 1
    labels:
      kustomize.toolkit.fluxcd.io/name: infrastructure-dynatrace
      kustomize.toolkit.fluxcd.io/namespace: flux-system
    name: dynasl1
    namespace: dynatrace
  spec:
    apiUrl: https://off#####.live.dynatrace.com/api
    tokens: dynakube
    activeGate:
      capabilities:
      - routing
      - kubernetes-monitoring
      - dynatrace-api
      resources:
        limits:
          cpu: 1000m
          memory: 1.5Gi
        requests:
          cpu: 500m
          memory: 512Mi
    oneAgent:
      classicFullStack:
        tolerations:
        - effect: NoSchedule
          key: node-role.kubernetes.io/master
          operator: Exists
        - effect: NoSchedule
          key: node-role.kubernetes.io/control-plane
          operator: Exists
        - effect: NoSchedule
          key: CriticalAddonsOnly
          operator: Exists
```

**Comment by luhi-DT** [comment]
Hi, thanks for reaching out. The input is valid and we would love to do that however if you read through the docs closely you will find a limitation for Operators and respectively their CSVs, which says that if they feature a conversion webhook they may only support AllNamespaces. That is the only reason why we have to choose AllNamespaces, as the Dynatrace Operator have a conversion webhook that converts DynaKube's from v1alpha1 to v1beta1. We already talked to RedHat about that, but currently it stands as it is and i can't give an outlook if or when this will change.

For reference: https://olm.operatorframework.io/docs/advanced-tasks/adding-admission-and-conversion-webhooks/#conversion-webhook-rules-requirements

Thanks!

**Comment by luhi-DT** [comment]
Thanks a lot for the detailed report! I will have a look on why this happened, currently it looks like only K8s OLM is the only affected platform that has this behaviour, all other releases and release platforms such as github, helm, RedHat marketplace, RedHat certified operators or RedHat community-operators are not affected and work as expected. 

I will keep you posted!

Thanks!

**Comment by ns-rkusumoto** [comment]
I found that some environment variables were not being set for oneagent.

oneagent:
  cloudNativeFullStack:
     env:
        - name: ONEAGENT_INSTALLER_SCRIPT_URL
          value: 
        - name: ONEAGENT_INSTALLER_DOWNLOAD_TOKEN
          value: 
        - name: ONEAGENT_INSTALLER_SKIP_CERT_CHECK
          value: "false"

image: docker.io/dynatrace/oneagent:1.77.1000

[setup docker oneagent](https://www.dynatrace.com/support/help/setup-and-configuration/setup-on-container-platforms/docker/set-up-dynatrace-oneagent-as-docker-container)

**Comment by luhi-DT** [comment]
That image is not supported by the operator. Therefore we don't set these.

**Comment by Jaykul** [comment]
And yet, not shipped in [0.12.1](https://github.com/Dynatrace/dynatrace-operator/compare/v0.12.0...v0.12.1)?
